### ISSUE TITLE
Cannot run breach: undefined symbol: g_type_class_adjust_private_offset

### ISSUE THREAD

**Issue by meznaric** [spolu/breach_core#137]
I'm running Elementary OS which is based on Ubuntu 12.04. After I fixed missing dependency problems **libharfbuzz.so.0** and **libudev.so.1** I get this error when running ./breach

```
./__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser: symbol lookup error:
./__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser: undefined symbol: g_type_class_adjust_private_offset
```

I tried [https://github.com/breach/breach_core/issues/60#issuecomment-48656278](https://github.com/breach/breach_core/issues/60#issuecomment-48656278) but it didn't help.

**Comment by spolu** [comment]
Which version of glib do you have installed?

**Comment by user12312** [comment]
I am having the same issue on linux mint 15 Olivia.

**Comment by meznaric** [comment]
I have glib-2.0 in my system.

**Comment by spolu** [comment]
Do you guys would like to go through the process of rebuilding it from Sources... takes some time but could be instructive as what is the cause for that error: https://github.com/breach/breach_core/wiki/Building-Breach-from-Source

**Comment by triclops200** [comment]
It is an glib issue, I had it last friday. The fix for me was upgrading to debian unstable which had a more modern version of glib. You may wish to try and link to an older version to support systems which use legacy glib.

**Comment by spolu** [comment]
Yep the build was done on Ubuntu 14.04 and should probably be done on a 12.04

**Comment by user12312** [comment]
Are there any beginner step by step instructions to get a build of this working? Seeing as the distribution on the website only works for some users.

**Comment by spolu** [comment]
Yes. Though it's quite long: https://github.com/breach/breach_core/wiki/Building-Breach-from-Source and from there see https://github.com/breach/exo_browser/wiki/Building-the-ExoBrowser

**Comment by user12312** [comment]
Thank you, I have seen those but to a non beginner they appear so vague. A Quick start with simple steps for non developers would be useful I think. That way people running 12.04 which is an LTS (long term support) distro and others based on it can get up and running and testing and helping to contribute to the community.

**Comment by spolu** [comment]
Hi guys,
I've made a build with Ubuntu 12.04 of the exobrowser. Can you try running it and tell me it that solves these problems: https://github.com/breach/releases/tree/master/exo_browser/v0.6.1916

**Comment by triclops200** [comment]
I can't comment about outdated systems because I'm on gentoo, but it has another bug: missing libudev.so.0. On Gentoo, at least, it's just libudev.so. It works once that bug's fixed, but I do have an up to date glib.

**Comment by user12312** [comment]
I can try it if you tell me where to place it. thanks!

**Comment by spolu** [comment]
@user12312 you can try running it right away... see if it solves the pb

**Comment by user12312** [comment]
sorry am a beginner which file do I replace in the main download?

**Comment by spolu** [comment]
You don't need to replace anything you can just download the exobrowser .tar.gz, unpack it and run the exobrowser executable in it

**Comment by user12312** [comment]
./exo_browser: error while loading shared libraries: libudev.so.0: cannot open shared object file: No such file or directory

**Comment by meznaric** [comment]
@spolu It's working with build you linked, but I had to manually link `libudev.so.1` to `libudev.so.0`. And then I had to run it with `--disable-setuid-sandbox`

**Comment by spolu** [comment]
This should be solved with the release of `v0.3.22-alpha.6`
Linux x64: http://bit.ly/1p8Jkmc

**Comment by spolu** [comment]
Make sure to check the README in the tarball or this page: https://github.com/breach/breach_core/wiki/Running-Breach-on-Linux
